docs/aws/audit/elbmonitoring/rules/elb_idle.mdx

```diff
@@ -24,6 +24,237 @@ CBP
+   ```sh
+   aws elb describe-load-balancers --query 'LoadBalancerDescriptions[*].LoadBalancerName'
+   ```
+
+2. **Check Load Balancer Metrics:**
+   For each load balancer, check the CloudWatch metrics to determine if they are receiving any traffic. You can use the `get-metric-statistics` command to fetch metrics like `RequestCount`.
+   ```sh
+   aws cloudwatch get-metric-statistics --namespace AWS/ELB --metric-name RequestCount --dimensions Name=LoadBalancerName,Value=<LoadBalancerName> --start-time <StartTime> --end-time <EndTime> --period 3600 --statistics Sum
+   ```
+
+3. **Tagging Load Balancers:**
+   Tag your load balancers with metadata that can help you identify their purpose and usage. This can help in future audits and automated scripts to identify idle resources.
+   ```sh
+   aws elb add-tags --load-balancer-names <LoadBalancerName> --tags Key=Environment,Value=Production Key=Owner,Value=TeamA
+   ```
+
+4. **Automate Monitoring:**
+   Set up a CloudWatch alarm to monitor the `RequestCount` metric for each load balancer. If the `RequestCount` is zero for a prolonged period, trigger an alert or an automated action.
+   ```sh
+   aws cloudwatch put-metric-alarm --alarm-name "IdleELBAlarm-<LoadBalancerName>" --metric-name RequestCount --namespace AWS/ELB --statistic Sum --period 3600 --threshold 1 --comparison-operator LessThanThreshold --dimensions Name=LoadBalancerName,Value=<LoadBalancerName> --evaluation-periods 24 --alarm-actions <SNS_Topic_ARN>
+   ```
+
+By following these steps, you can proactively monitor and manage your ELBs to ensure that no idle load balancers are present in your AWS environment.
+</Accordion>
+
+<Accordion title='Using Python'>
+To prevent idle Elastic Load Balancers (ELBs) in AWS using Python scripts, you can use the Boto3 library, which is the AWS SDK for Python. Here are the steps to achieve this:
+
+### Step 1: Install Boto3
+First, ensure you have Boto3 installed. You can install it using pip if you haven't already:
+```bash
+pip install boto3
+```
+
+### Step 2: Initialize Boto3 Client
+Initialize the Boto3 client for ELB:
+```python
+import boto3
+
+# Initialize a session using Amazon EC2
+session = boto3.Session(
+    aws_access_key_id='YOUR_AWS_ACCESS_KEY',
+    aws_secret_access_key='YOUR_AWS_SECRET_KEY',
+    region_name='YOUR_AWS_REGION'
+)
+
+# Initialize the ELB client
+elb_client = session.client('elb')
+```
+
+### Step 3: List All ELBs and Check for Idle ELBs
+Fetch all ELBs and check if they are idle (i.e., have no instances attached):
+```python
+def get_idle_elbs(elb_client):
+    idle_elbs = []
+    response = elb_client.describe_load_balancers()
+    
+    for elb in response['LoadBalancerDescriptions']:
+        if not elb['Instances']:
+            idle_elbs.append(elb['LoadBalancerName'])
+    
+    return idle_elbs
+
+idle_elbs = get_idle_elbs(elb_client)
+print(f"Idle ELBs: {idle_elbs}")
+```
```

Comment:
<b>Secret:</b> Secret Keyword <br/>
                    <b><a href="https://console.cloudanix.com/code?t=secrets" target='_blank'>Click here for more details</a></b>